typing/typecore.ml

```diff
@@ -4630,7 +4631,7 @@ and type_cases ?in_function env ty_arg ty_res partial_flag loc caselist =
   (* `Contaminating' unifications start here *)
   List.iter (fun f -> f()) !pattern_force;
   (* Post-processing and generalization *)
-  if propagate || erase_either then unify_pats (instance env ty_arg);
+  if partial <> None then unify_pats (instance env ty_arg);
```

Comment:
This seems OK too, but it only works because there is the `unify_pats ty_arg'` above. I.e. ty_arg may be really polymorphic.

Reply:
Fair point, I added the suggested call to `generalize`.